internal/datanode/data_sync_service.go

```diff
@@ -69,7 +69,8 @@ type dataSyncService struct {
 	timetickSender *timeTickSender        // reference to timeTickSender
 }
 
-func newDataSyncService(ctx context.Context,
+func newDataSyncService(
+	fgCtx, initCtx context.Context,
```

Comment:
Is it reasonable to have two ctx? 🤔

Reply:
yes, flowgraph ctx is used by flowgraphs, maintaining fg's lifetime, the child ctx of DataNode's ctx.

initCtx is just used when initializing a datasyncservice, will be canceled or timouted.

newDataSyncService is very complicated, I think we need to simplify it and split it to several function, then we can split the two ctx since they have different usages.

Reply:
linked a issue here: #27207

Reply:
`flowgraph` or `dataSyncService` shall be explictly closed by a `Close` call. Canceling a context is not a recommended way

Reply:
maybe we could remove the first ctx, since its not inuse actually

Reply:
> `flowgraph` or `dataSyncService` shall be explictly closed by a `Close` call. Canceling a context is not a recommended way

- Add initCtx is to enable canceled init progress;
- Remove fgCtx will be done when enhancing newDataSyncService
@jiaoew1991 @congqixia Can this PR move on?